public/js/Lineplot.js

```diff
@@ -0,0 +1,163 @@
+var d3 = require('d3');
+var _ = require('lodash');
+
+function Lineplot(sampleData, histogram) {
+    this.sampleData = sampleData;
+    this.featureLength = sampleData[0].feature_vector_std.length;
+    this.activeSample = 0;
+    this.activeFeature = 0;
+    this.histogram = histogram;
+
+    this.fullWidth = 600;
+    this.fullHeight = 400;
+
+    this.margin = {top: 20, right: 30, bottom: 20, left: 45};
+
+    this.width = this.fullWidth - this.margin.left - this.margin.right;
+    this.height = this.fullHeight - this.margin.top - this.margin.bottom;
```

Comment:
Yes, I like that. But maybe make them later arguments so they can have default values? Also, is it possible to define these as percentage of width with some minimum width value? And would that be considered good practice? =P